imap/imapd.c

```diff
@@ -6091,11 +6091,25 @@ static void cmd_search(char *tag, int usinguid)
                                 imapd_userisadmin || imapd_userisproxyadmin);
 
     /* Set FUZZY search according to config and quirks */
-    if (config_getswitch(IMAPOPT_SEARCH_FUZZY_ALWAYS) ||
-        (imapd_id.quirks & QUIRK_SEARCHFUZZY)) {
+    static const char *annot = IMAP_ANNOT_NS "search-fuzzy-always";
+    char *inbox = mboxname_user_mbox(imapd_userid, NULL);
+    struct buf val = BUF_INITIALIZER;
+    if (imapd_id.quirks & QUIRK_SEARCHFUZZY) {
+        /* Quirks overrule anything */
         searchargs->fuzzy_depth++;
     }
-
+    else if (!annotatemore_lookup(inbox, annot, imapd_userid, &val) && val.len) {
+        /* User may override global config */
+        if (config_parse_switch(buf_cstring(&val))) {
+            searchargs->fuzzy_depth++;
```

Comment:
-1 is a true value, so if the annotation is set to a junk value that the parser rejects, this `if()` will treat that the same as "yes"

If we want it to treat junk as "not-enabled", then this should be like `if(config_parse_switch(...) == 1)`, but I don't have an opinion about which behaviour we want.

Reply:
Though I see we validate attempts to set it, so I guess it shouldn't ever be junk at this point anyway

Reply:
Treating invalid values as true was on purpose. But now I come to realize that this was under the assumption, that search_fuzzy_always is enabled (as we do at Fastmail). I'll change this to use whatever is set in imapd.conf for invalid values.

Reply:
Fixed this in latest commit